src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -173,96 +124,68 @@ public AnomalyDetector(
-                ValidationAspect.DETECTOR
-            );
-        }
-        this.detectorId = detectorId;
-        this.version = version;
-        this.name = name;
-        this.description = description;
-        this.timeField = timeField;
-        this.indices = indices;
-        this.featureAttributes = features == null ? ImmutableList.of() : ImmutableList.copyOf(features);
-        this.filterQuery = filterQuery;
-        this.detectionInterval = detectionInterval;
-        this.windowDelay = windowDelay;
-        this.shingleSize = getShingleSize(shingleSize);
-        this.uiMetadata = uiMetadata;
-        this.schemaVersion = schemaVersion;
-        this.lastUpdateTime = lastUpdateTime;
-        this.categoryFields = categoryFields;
-        this.user = user;
-        this.detectorType = isMultientityDetector(categoryFields) ? MULTI_ENTITY.name() : SINGLE_ENTITY.name();
-        this.resultIndex = Strings.trimToNull(resultIndex);
-        String errorMessage = validateResultIndex(this.resultIndex);
-        if (errorMessage != null) {
-            throw new ValidationException(errorMessage, ValidationIssueType.RESULT_INDEX, ValidationAspect.DETECTOR);
-        }
-    }
 
-    public static String validateResultIndex(String resultIndex) {
-        if (resultIndex == null) {
-            return null;
-        }
-        if (!resultIndex.startsWith(CUSTOM_RESULT_INDEX_PREFIX)) {
-            return INVALID_RESULT_INDEX_PREFIX;
+        if (errorMessage != null && issueType != null) {
+            throw new ValidationException(errorMessage, issueType, ValidationAspect.DETECTOR);
+        } else if (errorMessage != null || issueType != null) {
+            throw new TimeSeriesException(CommonMessages.FAIL_TO_VALIDATE);
         }
-        if (resultIndex.length() > MAX_RESULT_INDEX_NAME_SIZE) {
-            return INVALID_RESULT_INDEX_NAME_SIZE;
-        }
-        if (!resultIndex.matches(RESULT_INDEX_NAME_PATTERN)) {
-            return INVALID_CHAR_IN_RESULT_INDEX_NAME;
-        }
-        return null;
+
+        this.detectorType = isHC(categoryFields) ? MULTI_ENTITY.name() : SINGLE_ENTITY.name();
     }
 
+    /*
+     * For backward compatiblity reason, we cannot use super class
+     * Config's constructor as we have detectionDateRange and
+     * detectorType that Config does not have.
+     */
     public AnomalyDetector(StreamInput input) throws IOException {
-        detectorId = input.readOptionalString();
+        id = input.readOptionalString();
         version = input.readOptionalLong();
         name = input.readString();
         description = input.readOptionalString();
         timeField = input.readString();
         indices = input.readStringList();
         featureAttributes = input.readList(Feature::new);
         filterQuery = input.readNamedWriteable(QueryBuilder.class);
-        detectionInterval = IntervalTimeConfiguration.readFrom(input);
+        interval = IntervalTimeConfiguration.readFrom(input);
```

Comment:
nit: should we maybe not change detectionInterval as we didn't in other parts of code cause of bwc. this is just naming and doesn't effect output though so up to you

Reply:
This variable is defined in its super class Config and meant to be used in both AnomalyDetector and Forecaster. So we need the naming change.

Reply:
got it, and then we just change name in json for bwc right?

Reply:
yes, we keep old name for bwc